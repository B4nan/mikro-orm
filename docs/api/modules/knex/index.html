<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.b3b658f68">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="MikroORM Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="MikroORM Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<link rel="preconnect" href="https://www.googletagmanager.com">
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-135618258-1"></script>
<script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-135618258-1",{})</script>
<link rel="search" type="application/opensearchdescription+xml" title="MikroORM" href="/opensearch.xml">
<script src="/js/custom.js"></script><title data-react-helmet="true">Module: knex | MikroORM</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="4.4"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-4.4"><meta data-react-helmet="true" property="og:title" content="Module: knex | MikroORM"><meta data-react-helmet="true" name="description" content="Module: knex"><meta data-react-helmet="true" property="og:description" content="Module: knex"><meta data-react-helmet="true" property="og:url" content="https://mikro-orm.io/docs/api/modules/knex"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="alternate" href="https://mikro-orm.io/docs/api/modules/knex" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://mikro-orm.io/docs/api/modules/knex" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link data-react-helmet="true" rel="canonical" href="https://mikro-orm.io/docs/api/modules/knex"><link rel="stylesheet" href="/assets/css/styles.234e9062.css">
<link rel="preload" href="/assets/js/styles.681ae21c.js" as="script">
<link rel="preload" href="/assets/js/runtime~main.cb4e39cc.js" as="script">
<link rel="preload" href="/assets/js/main.9cf1ac2e.js" as="script">
<link rel="preload" href="/assets/js/1.9db26fa1.js" as="script">
<link rel="preload" href="/assets/js/2.9b0b70c4.js" as="script">
<link rel="preload" href="/assets/js/d07d88d7.fcf9a057.js" as="script">
<link rel="preload" href="/assets/js/974.008a5939.js" as="script">
<link rel="preload" href="/assets/js/a831ca8d.3c29a1de.js" as="script">
<link rel="preload" href="/assets/js/17896441.6f040a3e.js" as="script">
<link rel="preload" href="/assets/js/7ae81572.31d13fda.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#main" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top navbarHideable_2qcr"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="MikroORM" class="themedImage_1VuW themedImage--light_3UqQ navbar__logo"><img src="/img/logo.svg" alt="MikroORM" class="themedImage_1VuW themedImage--dark_hz6m navbar__logo"><strong class="navbar__title"></strong></a><div class="navbar__item dropdown dropdown--hoverable dropdown--left"><a class="navbar__item navbar__link" href="/docs/installation">4.4</a><ul class="dropdown__menu"><li><a class="dropdown__link" href="/docs/next/api/modules/knex">Next</a></li><li><a aria-current="page" class="dropdown__link dropdown__link--active" href="/docs/api/modules/knex">4.4</a></li><li><a class="dropdown__link" href="/docs/4.3/installation">4.3</a></li><li><a class="dropdown__link" href="/docs/4.2/installation">4.2</a></li><li><a class="dropdown__link" href="/docs/4.1/installation">4.1</a></li><li><a class="dropdown__link" href="/docs/4.0/installation">4.0</a></li><li><a class="dropdown__link" href="/docs/3.6/installation">3.6</a></li><li><a class="dropdown__link" href="/docs/2.7/installation">2.7</a></li></ul></div><a class="navbar__item navbar__link" href="/docs/installation">Docs</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/docs/api">API</a><a class="navbar__item navbar__link" href="/docs/faq">FAQ</a><a class="navbar__item navbar__link" href="/blog">Blog</a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link" data-type="versions" href="/versions">latest: v4.4.2</a><a href="https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link" title="Chat on Slack">Slack</a><a href="https://github.com/mikro-orm/mikro-orm" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link" title="View on GitHub">GitHub</a><a href="https://twitter.com/MikroORM" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link" title="Twitter">Twitter</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_GrZ2"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span></button></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/"><img src="/img/logo.svg" alt="MikroORM" class="themedImage_1VuW themedImage--light_3UqQ navbar__logo"><img src="/img/logo.svg" alt="MikroORM" class="themedImage_1VuW themedImage--dark_hz6m navbar__logo"><strong class="navbar__title"></strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a role="button" class="menu__link menu__link--sublist">Versions</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/docs/next/api/modules/knex">Next</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active" href="/docs/api/modules/knex">4.4</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/4.3/installation">4.3</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/4.2/installation">4.2</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/4.1/installation">4.1</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/4.0/installation">4.0</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/3.6/installation">3.6</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/2.7/installation">2.7</a></li></ul></li><li class="menu__list-item"><a class="menu__link" href="/docs/installation">Docs</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/docs/api">API</a></li><li class="menu__list-item"><a class="menu__link" href="/docs/faq">FAQ</a></li><li class="menu__list-item"><a class="menu__link" href="/blog">Blog</a></li><li class="menu__list-item"><a class="menu__link" data-type="versions" href="/versions">latest: v4.4.2</a></li><li class="menu__list-item"><a href="https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA" target="_blank" rel="noopener noreferrer" class="menu__link" title="Chat on Slack">Slack</a></li><li class="menu__list-item"><a href="https://github.com/mikro-orm/mikro-orm" target="_blank" rel="noopener noreferrer" class="menu__link" title="View on GitHub">GitHub</a></li><li class="menu__list-item"><a href="https://twitter.com/MikroORM" target="_blank" rel="noopener noreferrer" class="menu__link" title="Twitter">Twitter</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_31aa"><main class="docMainContainer_3ufF"><div class="container padding-vert--lg docItemWrapper_3FMP"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div><span class="badge badge--secondary">Version: 4.4</span></div><div class="markdown"><h1><a aria-hidden="true" tabindex="-1" class="anchor" id="module-knex"></a>Module: knex<a class="hash-link" href="#module-knex" title="Direct link to heading">#</a></h1><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="table-of-contents"></a>Table of contents<a class="hash-link" href="#table-of-contents" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="namespaces"></a>Namespaces<a class="hash-link" href="#namespaces" title="Direct link to heading">#</a></h3><ul><li><a href="/docs/api/modules/knex.knex-1">Knex</a></li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="enumerations"></a>Enumerations<a class="hash-link" href="#enumerations" title="Direct link to heading">#</a></h3><ul><li><a href="/docs/api/enums/knex.querytype">QueryType</a></li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="classes"></a>Classes<a class="hash-link" href="#classes" title="Direct link to heading">#</a></h3><ul><li><a href="/docs/api/classes/knex.abstractsqlconnection">AbstractSqlConnection</a></li><li><a href="/docs/api/classes/knex.abstractsqldriver">AbstractSqlDriver</a></li><li><a href="/docs/api/classes/knex.abstractsqlplatform">AbstractSqlPlatform</a></li><li><a href="/docs/api/classes/knex.arraycriterianode">ArrayCriteriaNode</a></li><li><a href="/docs/api/classes/knex.criterianode">CriteriaNode</a></li><li><a href="/docs/api/classes/knex.criterianodefactory">CriteriaNodeFactory</a></li><li><a href="/docs/api/classes/knex.databaseschema">DatabaseSchema</a></li><li><a href="/docs/api/classes/knex.databasetable">DatabaseTable</a></li><li><a href="/docs/api/classes/knex.entitymanager">EntityManager</a></li><li><a href="/docs/api/classes/knex.entityrepository">EntityRepository</a></li><li><a href="/docs/api/classes/knex.objectcriterianode">ObjectCriteriaNode</a></li><li><a href="/docs/api/classes/knex.querybuilder">QueryBuilder</a></li><li><a href="/docs/api/classes/knex.querybuilderhelper">QueryBuilderHelper</a></li><li><a href="/docs/api/classes/knex.scalarcriterianode">ScalarCriteriaNode</a></li><li><a href="/docs/api/classes/knex.schemagenerator">SchemaGenerator</a></li><li><a href="/docs/api/classes/knex.schemahelper">SchemaHelper</a></li><li><a href="/docs/api/classes/knex.sqlentitymanager">SqlEntityManager</a></li><li><a href="/docs/api/classes/knex.sqlentityrepository">SqlEntityRepository</a></li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="interfaces"></a>Interfaces<a class="hash-link" href="#interfaces" title="Direct link to heading">#</a></h3><ul><li><a href="/docs/api/interfaces/knex.column">Column</a></li><li><a href="/docs/api/interfaces/knex.foreignkey">ForeignKey</a></li><li><a href="/docs/api/interfaces/knex.icriterianode">ICriteriaNode</a></li><li><a href="/docs/api/interfaces/knex.iquerybuilder">IQueryBuilder</a></li><li><a href="/docs/api/interfaces/knex.index">Index</a></li><li><a href="/docs/api/interfaces/knex.indexdef">IndexDef</a></li><li><a href="/docs/api/interfaces/knex.issame">IsSame</a></li><li><a href="/docs/api/interfaces/knex.joinoptions">JoinOptions</a></li><li><a href="/docs/api/interfaces/knex.knex-2">Knex</a></li><li><a href="/docs/api/interfaces/knex.table">Table</a></li><li><a href="/docs/api/interfaces/knex.tabledifference">TableDifference</a></li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="type-aliases"></a>Type aliases<a class="hash-link" href="#type-aliases" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="field"></a>Field<a class="hash-link" href="#field" title="Direct link to heading">#</a></h3><p>Æ¬ <strong>Field</strong>&lt;T&gt;: <em>string</em> | keyof T | <a href="/docs/api/modules/knex#knexstringref"><em>KnexStringRef</em></a> | <a href="/docs/api/classes/knex.knex.querybuilder"><em>QueryBuilder</em></a></p><h4><a aria-hidden="true" tabindex="-1" class="anchor" id="type-parameters"></a>Type parameters:<a class="hash-link" href="#type-parameters" title="Direct link to heading">#</a></h4><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td><code>T</code></td></tr></tbody></table><p>Defined in: <a href="https://github.com/mikro-orm/mikro-orm/blob/969d4229bd/packages/knex/src/typings.ts#L14" target="_blank" rel="noopener noreferrer">packages/knex/src/typings.ts:14</a></p><hr><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="knexstringref"></a>KnexStringRef<a class="hash-link" href="#knexstringref" title="Direct link to heading">#</a></h3><p>Æ¬ <strong>KnexStringRef</strong>: <a href="/docs/api/interfaces/knex.knex.ref"><em>Ref</em></a>&lt;<em>string</em>, { [alias: string]: <em>string</em>;  }&gt;</p><p>Defined in: <a href="https://github.com/mikro-orm/mikro-orm/blob/969d4229bd/packages/knex/src/typings.ts#L10" target="_blank" rel="noopener noreferrer">packages/knex/src/typings.ts:10</a></p><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="variables"></a>Variables<a class="hash-link" href="#variables" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="monkeypatchable"></a>MonkeyPatchable<a class="hash-link" href="#monkeypatchable" title="Direct link to heading">#</a></h3><p>â€¢ <code>Const</code> <strong>MonkeyPatchable</strong>: <em>object</em></p><h4><a aria-hidden="true" tabindex="-1" class="anchor" id="type-declaration"></a>Type declaration:<a class="hash-link" href="#type-declaration" title="Direct link to heading">#</a></h4><table><thead><tr><th>Name</th><th>Type</th><th>Value</th></tr></thead><tbody><tr><td><code>MySqlDialect</code></td><td><em>any</em></td><td><em>any</em></td></tr><tr><td><code>PostgresDialectTableCompiler</code></td><td><em>any</em></td><td><em>any</em></td></tr><tr><td><code>Sqlite3Dialect</code></td><td><em>any</em></td><td><em>any</em></td></tr><tr><td><code>TableCompiler</code></td><td><em>any</em></td><td><em>any</em></td></tr></tbody></table><p>Defined in: <a href="https://github.com/mikro-orm/mikro-orm/blob/969d4229bd/packages/knex/src/MonkeyPatchable.ts#L14" target="_blank" rel="noopener noreferrer">packages/knex/src/MonkeyPatchable.ts:14</a></p><h2><a aria-hidden="true" tabindex="-1" class="anchor" id="functions"></a>Functions<a class="hash-link" href="#functions" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor" id="knex"></a>Knex<a class="hash-link" href="#knex" title="Direct link to heading">#</a></h3><p>â€¢ <strong>Knex</strong>: </p><p>Defined in: node_modules/knex/types/index.d.ts:384</p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/mikro-orm/mikro-orm/edit/master/docs/versioned_docs/version-4.4/api/modules/knex.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-label="Edit page"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col text--right"><em><small>Last updated on <time datetime="2021-02-11T23:39:01.000Z" class="docLastUpdatedAt_XHNF">2/11/2021</time> by <strong>Renovate Bot</strong></small></em></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"></div><div class="pagination-nav__item pagination-nav__item--next"></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#table-of-contents" class="table-of-contents__link">Table of contents</a><ul><li><a href="#namespaces" class="table-of-contents__link">Namespaces</a></li><li><a href="#enumerations" class="table-of-contents__link">Enumerations</a></li><li><a href="#classes" class="table-of-contents__link">Classes</a></li><li><a href="#interfaces" class="table-of-contents__link">Interfaces</a></li></ul></li><li><a href="#type-aliases" class="table-of-contents__link">Type aliases</a><ul><li><a href="#field" class="table-of-contents__link">Field</a></li><li><a href="#knexstringref" class="table-of-contents__link">KnexStringRef</a></li></ul></li><li><a href="#variables" class="table-of-contents__link">Variables</a><ul><li><a href="#monkeypatchable" class="table-of-contents__link">MonkeyPatchable</a></li></ul></li><li><a href="#functions" class="table-of-contents__link">Functions</a><ul><li><a href="#knex" class="table-of-contents__link">Knex</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/installation">Installation &amp; Usage</a></li><li class="footer__item"><a href="https://github.com/mikro-orm/mikro-orm#-quick-start" target="_blank" rel="noopener noreferrer" class="footer__link-item">Quick Start</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/upgrading-v3-to-v4">Migration from v3 to v4</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/3.6/installation">Version 3.6 docs</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Community</h4><ul class="footer__items"><li class="footer__item"><a href="https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA" target="_blank" rel="noopener noreferrer" class="footer__link-item">Slack</a></li><li class="footer__item"><a href="https://stackoverflow.com/questions/ask?tags=mikro-orm" target="_blank" rel="noopener noreferrer" class="footer__link-item">Stack Overflow</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Social</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/blog">Blog</a></li><li class="footer__item"><a href="https://twitter.com/MikroOrm" target="_blank" rel="noopener noreferrer" class="footer__link-item">Twitter</a></li><li class="footer__item"><iframe src="https://ghbtns.com/github-btn.html?user=mikro-orm&amp;repo=mikro-orm&amp;type=star&amp;count=true" style="margin-top:10px" frameborder="0" scrolling="0" width="150" height="30" title="GitHub Stars"></iframe></li><li class="footer__item"><iframe src="https://ghbtns.com/github-btn.html?user=B4nan&amp;type=sponsor" frameborder="0" scrolling="0" width="150" height="30" title="Sponsor B4nan"></iframe></li></ul></div></div><div class="text--center"><div class="margin-bottom--sm"><img class="footer__logo" alt="MikroORM" src="/img/logo-header.svg"></div>Copyright Â© 2018-2021 Martin AdÃ¡mek. Built with Docusaurus. Icons made by <a href="https://www.flaticon.com/authors/surang" title="surang">surang</a> and <a href="https://www.flaticon.com/authors/skyclick" title="Skyclick">Skyclick</a>.</div></div></footer></div>
<script src="/assets/js/styles.681ae21c.js"></script>
<script src="/assets/js/runtime~main.cb4e39cc.js"></script>
<script src="/assets/js/main.9cf1ac2e.js"></script>
<script src="/assets/js/1.9db26fa1.js"></script>
<script src="/assets/js/2.9b0b70c4.js"></script>
<script src="/assets/js/d07d88d7.fcf9a057.js"></script>
<script src="/assets/js/974.008a5939.js"></script>
<script src="/assets/js/a831ca8d.3c29a1de.js"></script>
<script src="/assets/js/17896441.6f040a3e.js"></script>
<script src="/assets/js/7ae81572.31d13fda.js"></script>
</body>
</html>