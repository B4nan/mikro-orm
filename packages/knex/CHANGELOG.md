# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)


### Performance Improvements

* **core:** do not use `em.merge()` internally ([6a1a6d6](https://github.com/mikro-orm/mikro-orm/commit/6a1a6d68b65a20b8f1a78bf644844427f3b2dd1a))





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/knex





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)


### Bug Fixes

* **core:** refactor internals to reduce number of cycles ([#830](https://github.com/mikro-orm/mikro-orm/issues/830)) ([3994767](https://github.com/mikro-orm/mikro-orm/commit/3994767d93ef119d229bedffa77eb2ea3af5c775))
