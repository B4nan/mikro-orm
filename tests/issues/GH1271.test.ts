import { Collection, Entity, Logger, ManyToOne, MikroORM, OneToMany, PrimaryKey } from '@mikro-orm/core';
import { AbstractSqlDriver, SchemaGenerator } from '@mikro-orm/knex';


@Entity({ tableName: 'very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' })
class ChildEntity {

  @PrimaryKey()
  id!: number;

  @ManyToOne({ type: 'ParentEntity' })
  parent!: any;

}

@Entity()
class ParentEntity {

  @PrimaryKey()
  id!: number;

  @OneToMany({ entity: () => ChildEntity, mappedBy: 'parent' })
  children = new Collection<ChildEntity>(this);

}

describe('GH issue 1271', () => {
  let orm: MikroORM<AbstractSqlDriver>;
  const log = jest.fn();

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [ParentEntity, ChildEntity],
      dbName: `mikro_orm_test_gh_1271`,
      port: 3307,
      type: 'mysql',
      cache: { enabled: false },
    });
    const logger = new Logger(log, ['query', 'query-params']);
    Object.assign(orm.config, { logger });

    await new SchemaGenerator(orm.em).ensureDatabase();
    await orm.getSchemaGenerator().dropSchema();
  });

  afterAll(() => orm.close(true));

  test('Index and FK names should be a max of 64 chars in mysql', async () => {
    const sql = await orm.getSchemaGenerator().getCreateSchemaSQL();
    expect(sql).toMatchSnapshot();
    await orm.getSchemaGenerator().createSchema();
  });
});
