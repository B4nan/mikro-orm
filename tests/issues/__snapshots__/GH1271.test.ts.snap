// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GH issue 1271 Index and FK names should be a max of 64 chars in mysql 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`parent_entity\` (\`id\` int unsigned not null auto_increment primary key) default character set utf8mb4 engine = InnoDB;

create table \`very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\` (\`id\` int unsigned not null auto_increment primary key, \`parent_id\` int(11) unsigned not null) default character set utf8mb4 engine = InnoDB;
alter table \`very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\` add index \`very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxxxx_d90a7_index\`(\`parent_id\`);

alter table \`very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\` add constraint \`very_long_table_name_64_chars_xxxxxxxxxxxxxxxxxxxx_19cce_foreign\` foreign key (\`parent_id\`) references \`parent_entity\` (\`id\`) on update cascade;

set foreign_key_checks = 1;
"
`;
